src/org/zaproxy/zap/extension/frontEndScanner/FrontEndScannerAPI.java

```diff
@@ -0,0 +1,117 @@
+/*
+/* Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.frontEndScanner;
+
+import java.util.Locale;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+public class FrontEndScannerAPI extends ApiImplementor {
```

Comment:
Is this actually used?

Reply:
For `handleCallBack`?

Reply:
Ah, so the ACTION_GET_SCRIPTS isnt used - I think this should probably be removed then

Reply:
It is used [1], but so far the functionality seems to be a placeholder:

[1] https://github.com/zaproxy/zap-extensions/pull/1735/files/49a2cee7271e69942d1ad2cb877d97abfb7545d8#diff-2db6c8afdb9b68c38cdb09d567842a7dR107

Reply:
Yeah, uncleaned residue from my fiddling with the API.
There is no point exposing an API for now. The front-end javascript code is the one that need to communicate with ZAP, and I do that via the `handleCallBack`.

Reply:
The [`.getCallBackUrl`](https://github.com/zaproxy/zaproxy/blob/develop/src/org/zaproxy/zap/extension/api/API.java#L769) needs an `ApiImplementor` as first parameter.

Removing the overriding of `handleApiAction` and the related actions declarations.